managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -2661,6 +2708,34 @@ void setReadPosition(Position newReadPositionInt) {
         }
     }
 
+    void clearIncompleteAckedRecordsByLedgerId(final long ledgerId){
+        lock.writeLock().lock();
+        try {
+            List<Range> rangeListToDelete = individualDeletedMessages.asRanges().stream()
+                    .filter(range -> range.lowerEndpoint().getLedgerId() == ledgerId)
+                    .map(range -> Range.openClosed(
```

Comment:
Why can't we return the range direct?

Reply:
The solution has been revised. Could you take a look again？